org.lflang/src/org/lflang/FileConfig.java

```diff
@@ -434,34 +435,134 @@ public static void copyDirectory(Path src, Path dest) throws IOException {
         try {
-            Files.copy(src, dest, java.nio.file.StandardCopyOption.REPLACE_EXISTING);
+            Files.copy(source, destination, java.nio.file.StandardCopyOption.REPLACE_EXISTING);
         } catch (IOException e) {
             throw e;
         }
     }
+    
+    /**
+     *  Lookup a file in the classpath and copy its contents to a destination path 
+     *  in the filesystem.
+     * 
+     *  This also creates new directories for any directories on the destination
+     *  path that do not yet exist.
+     * 
+     *  @param source The source file as a path relative to the classpath.
+     *  @param destination The file system path that the source file is copied to.
+     * @throws IOException 
+     */
+    public void copyFileFromClassPath(String source, String destination) throws IOException {
+        InputStream sourceStream = this.getClass().getResourceAsStream(source);
+
+        if (sourceStream == null) {
+            throw new IOException(
+                "A required target resource could not be found: " + source + "\n" +
+                    "Perhaps a git submodule is missing or not up to date.\n" +
+                    "See https://github.com/icyphy/lingua-franca/wiki/downloading-and-building#clone-the-lingua-franca-repository.\n" +
+                    "Also try to refresh and clean the project explorer if working from eclipse.");
+        }
+
+        // Copy the file.
+        try {
+            // Make sure the directory exists
+            File destFile = new File(destination);
+            destFile.getParentFile().mkdirs();
+
+            Files.copy(sourceStream, Paths.get(destination), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException ex) {
+            throw new IOException(
+                "A required target resource could not be copied: " + source + "\n" +
+                    "Perhaps a git submodule is missing or not up to date.\n" +
+                    "See https://github.com/icyphy/lingua-franca/wiki/downloading-and-building#clone-the-lingua-franca-repository.",
+                ex);
+        } finally {
+            sourceStream.close();
+        }
+    }
+
+    /**
+     * Copy a list of files from a given source directory to a given destination directory.
+     * @param srcDir The directory to copy files from.
+     * @param dstDir The directory to copy files to.
+     * @param files The list of files to copy.
+     * @throws IOException 
+     */
+    public void copyFilesFromClassPath(String srcDir, String dstDir, List<String> files) throws IOException {
+        for (String file : files) {
+            copyFileFromClassPath(srcDir + '/' + file, dstDir + File.separator + file);
+        }
+    }
+    
+   /**
+    * Copy the 'fileName' from the 'srcDirectory' to the 'destinationDirectory'.
+    * This function has a fallback search mechanism, where if `fileName` is not
+    * found in the `srcDirectory`, it will try to find `fileName` via the following procedure:
```

Comment:
IMO, it should not look in the src directory, but the package root, meaning the parent of the src directory. This is the way it used to work, and I don't think we should change this.

Reply:
The reason being that not all files to copy are necessarily source files. They could be images. They could be database blobs.

Reply:
I didn't change how this works though, I think. The files directive has always [been](https://github.com/lf-lang/lingua-franca/blob/9e635d8d6405983c184550e7a6bd5d9269624a9a/org.lflang/src/org/lflang/generator/GeneratorBase.xtend#L482) relative to the [source file](https://github.com/lf-lang/lingua-franca/blob/9e635d8d6405983c184550e7a6bd5d9269624a9a/org.lflang/src/org/lflang/generator/GeneratorBase.xtend#L488) as far as I can remember. I agree that switching to the parent might be better. This is a relatively easy fix here, but I think we should have a discussion about it.